renku/cli/dataset.py

```diff
@@ -362,6 +374,108 @@ def remove(client, names):
     click.secho('OK', fg='green')
 
 
+@dataset.command('import')
+@click.argument('uri')
+@click.option('-n', '--name', help='Dataset name.')
+@click.option(
+    '-x',
+    '--extract',
+    is_flag=True,
+    help='Extract files before importing to dataset.'
+)
+@click.pass_context
+def import_(ctx, uri, name, extract):
+    """Import data from 3rd party provider."""
+    is_doi = DOIProvider.is_doi(uri)
+
+    if is_doi is False:
+        url = urlparse(uri)
+        if bool(url.scheme and url.netloc and url.params == '') is False:
+            raise BadParameter(
+                'Could not process {0}.\nValid formats are: DOI, URL'.
+                format(uri)
+            )
+
+    provider = None
+    if 'zenodo' in uri:
+        provider = ZenodoProvider()
+
+    try:
+        record = provider.find_record(uri, is_doi=is_doi)
+    except LookupError:
+        raise BadParameter(
+            'Could not process {0}.\nURI not found.'.format(uri)
+        )
+
+    files_ = record.get_files()
+    click.echo(
+        tabulate(
+            files_,
+            headers=OrderedDict((
+                ('checksum', None),
+                ('key', 'filename'),
+                ('size', None),
+                ('type', None),
+            ))
+        )
+    )
+
+    text_prompt = 'Do you wish to download?'
+    if record.last_version is False:
+        text_prompt = WARNING + 'Newer version found.\n' + text_prompt
+
+    if files_ and click.confirm(text_prompt):
+        data_folder = tempfile.mkdtemp()
+
+        pool = mp.Pool(mp.cpu_count())
+        processing = [
+            pool.apply_async(
+                download_file, args=(
+                    i,
+                    extract,
+                    data_folder,
+                    file_,
+                )
+            ) for i, file_ in enumerate(files_)
+        ]
+        for p in processing:
+            p.wait()
+        pool.close()
+
+        ctx.invoke(add, name=name or record.title, urls=[data_folder])
+        click.secho('OK', fg='green')
+
+
+def download_file(position, extract, data_folder, file, chunk_size=8192):
+    """Download file with progress tracking."""
+    url = file.links['self']
+    local_filename = Path(url.split('/')[-1])
```

Comment:
might be safer to use a combination of `urlparse` and `Path` methods, e.g.

`Path(urllib.parse.urlparse('https://path/to/file/').path)`